observer/probers/tls/tls_conf.go

```diff
@@ -0,0 +1,149 @@
+package probers
+
+import (
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/letsencrypt/boulder/observer/probers"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	certExpiryName = "obs_cert_expiry"
+	statusCodeName = "tls_prober_status_code"
+)
+
+// TLSConf is exported to receive YAML configuration.
+type TLSConf struct {
+	URL      string `yaml:"url"`
+	Root     string `yaml:"root"`
+	Response string `yaml:"response"`
+}
+
+// Kind returns a name that uniquely identifies the `Kind` of `Configurer`.
+func (c TLSConf) Kind() string {
+	return "TLS"
+}
+
+// UnmarshalSettings takes YAML as bytes and unmarshals it to the to an
+// TLSConf object.
+func (c TLSConf) UnmarshalSettings(settings []byte) (probers.Configurer, error) {
+	var conf TLSConf
+	err := yaml.Unmarshal(settings, &conf)
+	if err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+func (c TLSConf) validateURL() error {
+	url, err := url.Parse(c.URL)
+	if err != nil {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, expected a valid url", c.URL)
```

Comment:
Style nit: we try to only embed errors after `:` and the `: %s, err` should always be at the end of that error message. Also, let's return the `err`.

```suggestion
			"invalid 'url', got %q, expected a valid url: %s", c.URL, err)
```